curriculum/task_encoder.py

```diff
@@ -0,0 +1,124 @@
+from types import ModuleType
+import ast
+import inspect
+import json
+from typing import List
+import torch
+
+from tqdm import tqdm
+from transformers import AutoTokenizer, CodeGenModel
+
+
+def is_function_type(obj):
+  return inspect.isfunction(obj) and not inspect.isbuiltin(obj)
+
+def extract_module_fn(module: ModuleType):
+  fn_dict = {}
+  for name, fn in module.__dict__.items():
+    if is_function_type(fn) and not name.startswith('_'):
+      fn_dict[name] = fn
+  return fn_dict
+
+######################################################################
+# Task embedding assumptions
+# CHECK ME: currently assuming each agent has ONLY ONE task assigned during training,
```

Comment:
Correct -- will actually be one per team... still thinking about that

Reply:
a config/assert to ensure single task per agent was added to the env